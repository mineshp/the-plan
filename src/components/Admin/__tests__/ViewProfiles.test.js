import React from 'react';
import renderer from 'react-test-renderer';
import ViewProfilesComponent from '../ViewProfiles';
import { mockListProfiles } from '../../../helpers/test/testData/controlCentreData';

const mockHandleDeleteProfile = jest.fn();
const mockHandleProfileStatus = jest.fn();

describe('Admin > Manage > ViewProfiles', () => {
    const mockProfiles = mockListProfiles();
    it('renders ListUsers component', () => {
        const tree = renderer.create(
            <ViewProfilesComponent
                profiles={mockProfiles}
                handleDeleteProfile={mockHandleDeleteProfile}
                handleProfileStatus={mockHandleProfileStatus}
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
