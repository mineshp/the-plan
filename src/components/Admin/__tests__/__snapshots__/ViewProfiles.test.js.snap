// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin > Manage > ViewProfiles renders ListUsers component 1`] = `
<div
  className="ui container"
>
  <p>
    Existing profiles that have been setup.
  </p>
  <table
    className="ui celled striped table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
        >
          Name
        </th>
        <th
          className=""
        >
          Status
        </th>
        <th
          className=""
        >
          Update
        </th>
        <th
          className=""
        >
          Delete
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    >
      <tr
        className=""
      >
        <td
          className=""
        >
          MIN_JIGS_WEDDING
        </td>
        <td
          className="activeProfileColour"
        >
          ACTIVE
        </td>
        <td
          className="collapsing"
        >
          <a
            className="ui teal icon button"
            disabled={undefined}
            onClick={[Function]}
            profile={
              Object {
                "_id": "1234567",
                "active": true,
                "name": "MIN_JIGS_WEDDING",
              }
            }
            role="button"
            tabIndex={undefined}
            title="Profile active"
          >
            <i
              aria-hidden="true"
              className="lock icon"
            />
          </a>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui pink icon button"
            disabled={undefined}
            id="1234567"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            title="Delete Profile"
          >
            <i
              aria-hidden="true"
              className="trash icon"
            />
          </button>
        </td>
      </tr>
      <tr
        className=""
      >
        <td
          className=""
        >
          HOME_DIY
        </td>
        <td
          className="activeProfileColour"
        >
          ACTIVE
        </td>
        <td
          className="collapsing"
        >
          <a
            className="ui teal icon button"
            disabled={undefined}
            onClick={[Function]}
            profile={
              Object {
                "_id": "1234568",
                "active": true,
                "name": "HOME_DIY",
              }
            }
            role="button"
            tabIndex={undefined}
            title="Profile active"
          >
            <i
              aria-hidden="true"
              className="lock icon"
            />
          </a>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui pink icon button"
            disabled={undefined}
            id="1234568"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            title="Delete Profile"
          >
            <i
              aria-hidden="true"
              className="trash icon"
            />
          </button>
        </td>
      </tr>
      <tr
        className="negative"
      >
        <td
          className=""
        >
          CHRISTMAS
        </td>
        <td
          className="inActiveProfileColour"
        >
          DISABLED
        </td>
        <td
          className="collapsing"
        >
          <a
            className="ui violet active icon button"
            disabled={undefined}
            onClick={[Function]}
            profile={
              Object {
                "_id": "1234569",
                "active": false,
                "name": "CHRISTMAS",
              }
            }
            role="button"
            tabIndex={undefined}
            title="Profile disabled"
          >
            <i
              aria-hidden="true"
              className="unlock icon"
            />
          </a>
        </td>
        <td
          className="collapsing"
        >
          <button
            className="ui pink icon button"
            disabled={undefined}
            id="1234569"
            onClick={[Function]}
            role="button"
            tabIndex={undefined}
            title="Delete Profile"
          >
            <i
              aria-hidden="true"
              className="trash icon"
            />
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
